api-tests/src/test/java/org/openmrs/module/reporting/report/renderer/TextTemplateRendererTest.java

```diff
@@ -129,4 +134,61 @@ public ReportDesign getDesign(String argument) {
 		xml = templateType != null ? StringUtils.deleteWhitespace(xml) : "Males=2Females=2";
 		Assert.assertEquals(xml, renderedOutput);
 	}
+
+	private void shouldRenderJSONTemplate(String templateName, String templateType) throws Exception {
+
+		ReportDefinition reportDefinition = new ReportDefinition();
+		reportDefinition.setName("Test Report");
+
+		GenderCohortDefinition males = new GenderCohortDefinition();
+		males.setName("Males");
+		males.setMaleIncluded(true);
+
+		GenderCohortDefinition females = new GenderCohortDefinition();
+		females.setName("Females");
+		females.setFemaleIncluded(true);
+
+		CohortCrossTabDataSetDefinition genderDsd = new CohortCrossTabDataSetDefinition();
+		genderDsd.addColumn("males", males, null);
+		genderDsd.addColumn("females", females, null);
+		reportDefinition.addDataSetDefinition("genders", genderDsd, null);
+
+		final ReportDesign reportDesign = new ReportDesign();
+		reportDesign.setName("TestDesign");
+		reportDesign.setReportDefinition(reportDefinition);
+		reportDesign.setRendererType(JSONTemplateRenderer.class);
+
+		if (templateType != null) {
+			reportDesign.addPropertyValue(JSONTemplateRenderer.TEMPLATE_TYPE, templateType);
+		}
+
+		EvaluationContext context = new EvaluationContext();
+		ReportDefinitionService rs = Context.getService(ReportDefinitionService.class);
+		ReportData reportData = rs.evaluate(reportDefinition, context);
+
+		ReportDesignResource resource = new ReportDesignResource();
+		resource.setName(templateName);
+		InputStream is = OpenmrsClassLoader.getInstance().getResourceAsStream("org/openmrs/module/reporting/report/renderer/" + templateName);
+		resource.setContents(IOUtils.toByteArray(is));
+		IOUtils.closeQuietly(is);
+		reportDesign.addResource(resource);
+
+		JSONTemplateRenderer renderer = new JSONTemplateRenderer() {
+			public ReportDesign getDesign(String argument) {
+				return reportDesign;
+			}
+		};
+
+		ByteArrayOutputStream baos = new ByteArrayOutputStream();
+		renderer.render(reportData, "ReportData", baos);
+		String renderedOutput = StringUtils.deleteWhitespace(baos.toString());
+
+		String json ="{\n" +
+				"  \"Males\": \"#genders.males#\",\n" +
```

Comment:
@solemabrothers Add a test for non Integer decimal value to compare how it will be rendered - this may be a bug in the Text Render too